### DISCUSSION TITLE
Tracing of Async Tasks in Agentic Workflow

### DISCUSSION THREAD

**Discussion by ashish-vwo** [langfuse/langfuse#4358] [Support]
Hello Langfuse Community,
I'm currently working on a project with Agents and have encountered an issue. When executing tasks asynchronously, I'm unable to trace the OpenAI calls made by agents. I would appreciate any insights or suggestions on how to resolve this.
Thanks and Regards,
Ashish

**Comment by marcklingen** [comment]
Can you share more about your implementation? Happy to help as the openai integration should work on all async calls.

**Comment by ashish-vwo** [comment]
I have 5 tasks that are executed in the following order:
Task 1 → (Tasks 2, 3, and 4 executed asynchronously) → Task 5.

Currently, I can capture the OpenAI API calls for Task 1 and Task 5, but for Tasks 2, 3, and 4, I am only able to record the spans, not the actual OpenAI generation calls. This is critical for debugging the workflow and improving my system. I am capturing the traces via observe decorator for langchain.

**Reply by marcklingen** [reply]
Happy to have a look, can you share a minimum reproduction example, e.g. via a Google Colab / notebook?
